docs: Add contributing guidelines (#157)
kernelci/kernelci-project: kernelci.org/content/en/docs/contrib.md

---
title: "Contributing Guidelines"
date: 2022-12-08T09:21:56Z
draft: false
weight: 2
---

KernelCI core project is open for contributions. Contributions may consist of

> adding new build, tests and device types as well as features and bugfixes for

Reviewer: Should it be plural?

```suggestion
adding new builds, tests and device types as well as features and bugfixes for
```

KernelCI core tools.
The best way to contribute is to send a PR to [kernelci-core](https://github.com/kernelci/kernelci-core).

> When the PR is created, [KernelCI staging](https://kernelci.org/docs/instances/staging)

Reviewer: ```suggestion
When the PR is created, the [KernelCI staging](https://kernelci.org/docs/instances/staging)
```

instance takes care of updating the [staging.kernelci.org branch](https://github.com/kernelci/kernelci-core/tree/staging.kernelci.org).
In general the branch is updated every 8h and a limited set of builds and tests

> is run on it. More detailed information about the logic behind staging runs can

Reviewer: ```suggestion
are run on it. More detailed information about the logic behind staging runs can
```

Author: I'm not sure if it should be **are** I'm mentioning a _set_ here, that's why I used **is**.  Having said that I'm not a native English speaker so I might be wrong. @gctucker maybe you could help?

Reviewer 2: Even if the noun is singular, when it designates a plural entity it should be treated as plural e.g. "the team are on track to win the championship".  So it should be "are" imho.

Reviewer: I think either is fine, as far as I understand it depends on what one wants to emphasize, and singular is more common in American English (https://en.wikipedia.org/wiki/American_and_British_English_grammatical_differences#Subject-verb_agreement).

> be found [here](https://kernelci.org/docs/instances/staging).

Reviewer: ```suggestion
be found in the [staging instance documentation](https://kernelci.org/docs/instances/staging).
```

Reviewer 2: I think `here` links are bad, the text should be informative regardless of any hyperlinks.  That's some web design "best practice" rule I learnt a while ago...  So I'd rather leave `the staging instance documentation`.

Reviewer: Yeah, exactly (though I am not sure if you are agreeing or perhaps you read the diff the other direction).

Reviewer 2: Ah yes sorry, my bad :coffee:

> There are several guidelines which can facilitate the PR review process.

Reviewer: ```suggestion
There are several guidelines which can facilitate the PR review process:
```

> 1. Make sure the PR is well described

Reviewer 3: We are missing a explanation about KernelCI usage of DCO (Signed-off-by: reviewed-by: tested-by:) in comments.

Author: Is it OK to add it in as a follow-up PR?

Reviewer 3: ok

Reviewer 2: @mgalka Maybe create a GitHub issue so we don't forget about this by merging this PR?

Reviewer 2: Have you created an issue for this?

Reviewer 2: @mgalka See above - I haven't found any issue. I think that's the only thing to resolve before it can be merged

   1. Describe the purpose of the changes
   2. Example use cases are welcome

> 2. Attach staging build/test results when possible

Reviewer 3: Use a permanent link for staging build/test results when possible, as current links get removed after few weeks.
maybe suggest pastebin ( like github gist )

Reviewer 4: IMHO it is a bit hard to pastebin results from dashboard, where we have heavy ajax. Probably it is acceptable to have expiring link for new tests PR and just as confirmation that all works as designed.
There is very small chance we will need to review this test results.
But i agree that we need permanent links in case of discovered real bugs or commits.

Reviewer 3: > where we have heavy ajax.
I'm talking mostly about logs that are on https://storage.staging.kernelci.org/ and https://storage.kernelci.org/ that are temporary

Reviewer 3: just use something like wgetpastebin

Reviewer 2: Staging builds and tests aren't permanent.

Things we can treat as permanent are mailing list archives, anything on GitHub (issues, PRs etc.) and in theory results on the production dashboard although that's more a grey area as they might disappear after a while too.

I think it's best if we can just mention some log excerpts in the commit messages directly inline, and use `Link:` for email threads like in the kernel git history.

>    1. Staging instance is updated every 8h. If the PR is expected to produce build/test results

Reviewer: The first sentence is already explained above -- should it be removed here?

Reviewer: The GitHub UI makes my comment hard to understand -- I am referring here to line 24 but GitHub shows context, i.e. in particular I was talking about the "Staging instance is updated every 8h." sentence.

Author: I put it there as a reminder, just to make it clear that the author needs to wait at least 8h to see results on staging. On second thought, It may be too much.

Reviewer: Yeah, I think either way is fine -- I didn't know if it was intentional. In some types of material, there is value on repeating key information/concepts.

One alternative is saying something shorter that reminds the reader about it, without duplicating information. For instance, something along the lines of "If build/test results are expected to be visible, then link them from staging." (which also allows you to just say "If not" in the next line, if I understand correctly)

>    1. Staging instance is updated every 8h. If the PR is expected to produce build/test results
>    check [staging dashboard](https://staging.kernelci.org) and make sure these are mentioned in the PR comment

Reviewer: Perhaps an example of how the ideal PR comments and what sort of link to staging is expected would be useful. In particular, the links may help to understand the kind of URLs/paths expected.

Also:

```suggestion
   1. The staging instance is updated every 8h. If the PR is expected to produce build/test results, then
   check [staging dashboard](https://staging.kernelci.org) and make sure these are mentioned in the PR comment
```

Author: Actually, I looked for a PR that could be put here as an example, but all I looked into were using links to logs on stored on staging. I decided to not put it here as it was agreed that we're going to either paste log excerpts to the comments or feed data to  pastebin and paste links from there.

   2. If the results are not visible, but they should be, mention it in the comments, too
3. Make sure that reviewers' comments and questions are addressed
   1. When there are comments unanswered for more than 1 month the PR will be closed
4. In case there is a need to consult the PR with KernelCI maintainers join the open hours
   1. Open hours take place every Thursday at 12:00 UTC at KernelCI [Jitsi](https://meet.kernel.social/kernelci-dev)
5. You can reach KernelCI maintainers also at:
   1. IRC - #kernelci channel on [libera.chat](https://libera.chat/)

>    2. KernelCI mailing list - [kernelci@lists.linux.dev](mailto:kernelci@lists.linux.dev)

Reviewer: To avoid duplication, perhaps it is best to link to a page containing these contact details?

Reviewer 2: Namely `/docs/org/maintainers/`.
